File: python/src/get_account_example.py
"""
get_account_info_example demonstrates the required operation to retrieve account information from the Libra blockchain
"""

from cryptography.hazmat.primitives.asymmetric.ed25519 import Ed25519PrivateKey
from diem import testnet, AuthKey, utils


def main():
    # connect to testnet
    client = testnet.create_client()

    # generate private key
    private_key = Ed25519PrivateKey.generate()
    # generate auth key
    auth_key = AuthKey.from_public_key(private_key.public_key())
    print(f"Generated address: {utils.account_address_hex(auth_key.account_address())}")

    # create account
    faucet = testnet.Faucet(client)
    testnet.Faucet.mint(faucet, auth_key.hex(), 100_000_000, "XUS")

    # get account information
    account = client.get_account(auth_key.account_address())

    print("Account info:")
    print(account)


if __name__ == "__main__":
    main()

File: typescript/src/getAccountExample.ts
import {bytesToHexString} from "@libra/client/dist/utils/bytes";
import {Signer} from "@libra/client/dist/utils/signer";
import AccountKeys from "@libra/client/dist/account/accountKeys";
import {TestnetClient} from "@libra/client";

/**
 * getAccountExample demonstrates the required operation to retrieve account information from the Libra blockchain
 */

async function main() {
  //generate keys
  const keyPair = Signer.generateKeyPair();
  const accountKeys = new AccountKeys(keyPair);

  //connect to client
  const client = new TestnetClient();

  //create account
  await client.mint(bytesToHexString(accountKeys.authKey), BigInt(1340000000), "Coin1")

  //get account information
  const account = await client.getAccount(accountKeys.accountAddress)

  console.log("~ Account info: %O", account);
}

main();
